presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveStorageFormats.java

```diff
@@ -1,8 +1,284 @@
+        String insertInto = format("INSERT INTO %s " +
+                "SELECT " +
+                "orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, " +
+                "linestatus, shipinstruct, shipmode, comment, returnflag " +
+                "FROM tpch.%s.lineitem", tableName, TPCH_SCHEMA);
+        query(insertInto);
+
+        // SELECT FROM TABLE
+        assertSelect("select sum(tax), sum(discount), sum(length(returnflag)) from %s", tableName);
+
+        // DROP TABLE
+        query(format("DROP TABLE %s", tableName));
+    }
+
+    @Test(dataProvider = "storage_formats", groups = {STORAGE_FORMATS})
+    public void testCreatePartitionedTableAs(StorageFormat storageFormat)
+    {
+        setSessionProperties(storageFormat);
+
+        String tableName = "storage_formats_test_create_table_as_select_partitioned_" + storageFormat.getName().toLowerCase();
+
+        // DROP TABLE
+        query(format("DROP TABLE IF EXISTS %s", tableName));
+
+        // CREATE TABLE AS SELECT
+        String createTableAsSelect = format(
+                "CREATE TABLE %s WITH (format='%s', partitioned_by = ARRAY['returnflag']) AS " +
+                        "SELECT " +
+                        "orderkey, partkey, suppkey, linenumber, quantity, extendedprice, discount, tax, " +
+                        "linestatus, shipinstruct, shipmode, comment, returnflag " +
+                        "FROM tpch.%s.lineitem",
+                tableName,
+                storageFormat.getName(),
+                TPCH_SCHEMA);
+        query(createTableAsSelect);
+
+        // SELECT FROM TABLE
+        assertSelect("select sum(tax), sum(discount), sum(length(returnflag)) from %s", tableName);
+
+        // DROP TABLE
+        query(format("DROP TABLE %s", tableName));
+    }
+
+    private static void assertSelect(String query, String tableName)
+    {
+        QueryResult expected = query(format(query, "tpch." + TPCH_SCHEMA + ".lineitem"));
+        List<Row> expectedRows = expected.rows().stream()
+                .map((columns) -> row(columns.toArray()))
+                .collect(toImmutableList());
+        QueryResult actual = query(format(query, tableName));
+        assertThat(actual)
+                .hasColumns(expected.getColumnTypes())
+                .containsExactly(expectedRows);
+    }
+
+    private static void setSessionProperties(StorageFormat storageFormat)
+    {
+        setSessionProperties(storageFormat.getSessionProperties());
+    }
+
+    private static void setSessionProperties(Map<String, String> sessionProperties)
+    {
+        Connection connection = defaultQueryExecutor().getConnection();
+        try {
+            PrestoConnection prestoConnection = connection.unwrap(PrestoConnection.class);
+            // create more than one split
+            prestoConnection.setSessionProperty("task_writer_count", "4");
+            prestoConnection.setSessionProperty("redistribute_writes", "false");
```

Comment:
Seems a bit strange to hard code these into the test instead of setting them in the server startup.